packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
 import 'package:meta/meta.dart';
 
-import 'print.dart';
+/// Styles for displaying a node in a [DiagnosticsNode] tree.
+///
+/// See also:
+///
+///  * [DiagnosticsNode.toStringDeep], which dumps text art trees for these
+///    styles.
+///  * [_normalTextConfiguration], which shows what text art for the standard
+///    style looks like.
+enum DiagnosticsTreeStyle {
+
+  /// Normal style for displaying trees.
+  normal,
```

Comment:
do you have any perferences for the name? I could make the styles extremely implementation specific and make this style
DiagnosticsTreeStyle.renderObject
and make the dense style
DiagnosticsTreeStyle.widget
On the other hand, I think this is legitimately a pretty standard middle of the road default/standard/normal style with nothing surprising. For example, I would recommend that a new tree of diagnostics nodes use this style barring extra information.